### PR TITLE
Build: seed project setup

### PR DESCRIPTION
**Pull request by omarqx**
## Summary of change

Create basic project structure.

## Change breakdown

- create seed npm project
- setup typescript
- setup babel
- setup linting using eslint and prettier
- setup testing using jest
- add vscode debugging setup
- add github templates for issues and PRs
- add sample code to check everything works

## How to test

`npm install`

`npm test`

`npm run lint`

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,3 +1,15 @@
+## Getting started
+
+1. clone repo
+
+2. install dependencies
```
Comment (ahmadrezatabibi) [inline README.md:5]: nit: in the future, we can create a page for the dependencies and step by step start off the project

File: package.json [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+    "test": "jest ./src --coverage",
+    "lint": "eslint ./src --ext ts,js"
+  },
+  "repository": {
+    "type": "git",
+    "url": "git+https://github.com/InterviewDojo/Algorithms.git"
+  },
+  "keywords": [],
+  "author": "",
+  "license": "MIT",
```
Comment (ahmadrezatabibi) [inline package.json:15]: never done this myself, but did we need to do anything for the copyright and licensing of the project?

**Comment by ahmadrezatabibi** [comment]
LGTM, thanks man 🙌

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -0,0 +1,14 @@
+name: CI
```
Comment (ahmadrezatabibi) [inline .github/workflows/ci.yml:1]: 👍 never done that setup myself awesome dude

File: package.json (hunk 2 above)
Comment (omarqx) [reply package.json:15]: https://github.com/git/git-scm.com/blob/main/MIT-LICENSE.txt

no liability

**Pull request merged**